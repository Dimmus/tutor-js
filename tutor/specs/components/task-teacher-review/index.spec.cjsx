EventRow = require '../../../src/components/student-dashboard/event-row'
{Testing, _, SnapShot, React} = require '../helpers/component-testing'

STATS = require '../../../api/plans/1/stats.json'
REVIEW = require '../../../api/plans/1/review.json'

{TaskPlanStatsStore, TaskPlanStatsActions} = require '../../../src/flux/task-plan-stats'
{TaskTeacherReviewStore, TaskTeacherReviewActions} = require '../../../src/flux/task-teacher-review'

{TaskTeacherReview} = require '../../../src/components/task-teacher-review'

describe 'Task Teacher Review', ->
  beforeEach ->
    TaskPlanStatsActions.loaded('1', STATS)
    TaskTeacherReviewActions.loaded('1', REVIEW)
    @props =
      courseId: '1'
      id: '1'

  it 'renders and matches snapshot', ->
    review = shallow(<TaskTeacherReview {...@props} />)
    expect(review).toHaveRendered('ReviewShell[id="1"]')
    expect(review).toHaveRendered('StatsModalShell[id="1"]')
    expect(SnapShot.create(
      <TaskTeacherReview {...@props} />
    ).toJSON() ).toMatchSnapshot()
    undefined
