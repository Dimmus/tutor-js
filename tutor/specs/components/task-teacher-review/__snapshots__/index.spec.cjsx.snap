exports[`Task Teacher Review renders and matches snapshot 1`] = `
<div
  className="pinned-container task-teacher-review"
  style={
    Object {
      "marginTop": "30px",
    }
  }>
  <div
    className="container-fluid">
    <div
      className="row">
      <div
        className="col-sm-8">
        <div
          className="loadable is-loading">
          Loading... 
        </div>
      </div>
      <div
        className="col-sm-4">
        <div
          className="loadable is-loading">
          Loading... 
        </div>
      </div>
    </div>
  </div>
</div>
`;
